test/Constraints/diagnostics.swift

```diff
@@ -82,6 +83,15 @@ let a = A()
 for j in i.wibble(a, a) { // expected-error {{type 'A' does not conform to protocol 'SequenceType'}}
 }
 
+// Generic as part of function/tuple types
```

Comment:
Add tests involving typealiases and ensure the diagnostics are still useful and we don't fall back to the general path, and that the diagnostic involves the sugared typealias type and not what it points to